src/main/utils/resource.ts

```diff
@@ -1,112 +1,112 @@
+        'https://releases.jfrog.io/artifactory',
+        '',
+        '',
+        '',
+        ''
+    );
 
-//     public async download() {
-//         if (!fs.existsSync(this.TMP_DOWNLOAD_PATH)) {
-//             fs.mkdirSync(this.TMP_DOWNLOAD_PATH, { recursive: true });
-//         } else if (Date.now() - fs.statSync(this.TMP_DOWNLOAD_PATH).birthtimeMs <= Resource.MILLISECONDS_IN_HOUR) {
-//             // By here, someone else is already downloading the scanner.
-//             return;
-//         } else {
-//             // Seems like it is a left over from other download.
-//             fs.rmSync(this.TMP_DOWNLOAD_PATH);
-//         }
-//         try {
-//             if (this._connectionManager !== undefined) {
-//                 await this._connectionManager.downloadArtifactToFile(this.resourceRemotePath, path.join(this.TMP_DOWNLOAD_PATH, this.resourceName));
-//             } else {
-//                 await this._defaultJfrogClient
-//                     .artifactory()
-//                     .download()
-//                     .downloadArtifactToFile(this.resourceRemotePath, path.join(this.TMP_DOWNLOAD_PATH, this.resourceName));
-//             }
+    constructor(
+        private _homeDir: string,
+        private downloadSource: string,
+        resourceName: string,
+        private _logManager: LogManager,
+        connectionManager?: JfrogClient
+    ) {
+        this.downloadDir = path.join(_homeDir, 'download');
+        this.downloadTarget = path.join(this.downloadDir, resourceName);
+        this.path = path.join(_homeDir, resourceName);
+        if (connectionManager !== undefined) {
+            // Override the default connection manager.
+            this._connectionManager = connectionManager;
+        }
+    }
 
-//             await this.copy(path.join(this.TMP_DOWNLOAD_PATH, this.resourceName), path.join(this.resourceLocalPath, this.resourceName));
-//         } catch (error) {
-//             this._logManager.logMessage('failed to update the applicable scanner: ' + error, 'WARN');
-//         } finally {
-//             fs.rmSync(this.TMP_DOWNLOAD_PATH, { recursive: true, force: true });
-//         }
-//     }
-//     public copy(oldPath: string, newPath: string) {
-//         return new Promise((resolve, reject) => {
-//             const readStream: fs.ReadStream = fs.createReadStream(oldPath);
-//             const writeStream: fs.WriteStream = fs.createWriteStream(newPath);
+    public get homeDir(): string {
+        return this._homeDir;
+    }
 
-//             readStream.on('error', err => reject(err));
-//             writeStream.on('error', err => reject(err));
+    // Update the resource to the latest released
+    public async update(withExecPrem: boolean) {
+        const updateAvailable: boolean = await this.isUpdateAvailable();
+        if (!updateAvailable) {
+            return;
+        }
+        this._logManager.logMessage('Downloading new update from ' + this.downloadTarget, 'DEBUG');
```

Comment:
I think the user has the right to know what are we downloading for him

Reply:
`this._logManager.logMessage('Downloading CVE Applicability scanner from ' + this.downloadTarget, 'DEBUG');`
WDYT?